### PR TITLE
Use useResourcePermissions in block-library and the widgets editor

### PR DESCRIPTION
**Pull request by adamziel**
## What?
Applies the [newly stabilized useResourcePermissions](https://github.com/WordPress/gutenberg/pull/38785) hook to other places in the repository.

### CODE REVIEW

File: packages/block-library/src/navigation-submenu/edit.js [hunk 1]
```diff
@@ -294,15 +297,18 @@ export default function NavigationSubmenuEdit( {
 	const itemLabelPlaceholder = __( 'Add text…' );
 	const ref = useRef();
 
+	const [ , { canCreate: userCanCreatePages } ] =
+		useResourcePermissions( 'pages' );
```
Comment (gziolo) [inline packages/block-library/src/navigation-submenu/edit.js]: This part isn't the most intuitive with unused first param. What was the reason `hasResolved` wasn't merged with all other return values?

The code example included in the package presents the example that fits better here:

https://github.com/WordPress/gutenberg/blob/2026f13a17381f5bf36c84b141db071ccb27a028/packages/core-data/src/hooks/use-resource-permissions.ts#L50

Was it a sign? 😄

Comment (adamziel) [reply packages/block-library/src/navigation-submenu/edit.js]: I agree, I just updated the function signature

**Comment by gziolo** [comment]
Code changes look good. I'm wondering what would be the best way to test changes applied, any ideas?

The refined API for the use cases covered in this PR looks way better than before:

```js
const { canCreate: userCanCreatePages } = useResourcePermissions( 'pages' );
```

The variable aliasing is still something that isn't ideal, but it could also be coded in an alternative way:

```js
const pagesPermissions = useResourcePermissions( 'pages' );
// then in the code
pagesPermissions.canCreate ? 'yes' : 'no';
```

I guess it gives enough flexibility for start 👍🏻

**Comment by adamziel** [comment]
@gziolo I like your suggestion and I added these names as you proposed.

**Review by gziolo — APPROVED** [review]

**Pull request merged**